Plugin/Controller/Cart/Add.php

```diff
@@ -0,0 +1,195 @@
+     * @param \Magento\Framework\Controller\Result\RedirectFactory $redirectFactory
+     * @param \DevStone\UsageCalculator\Model\ResourceModel\MaxUsage\CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        \Magento\Customer\Model\Session $session,
+        \Magento\Framework\App\RequestInterface $request,
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollectionFactory,
+        \Magento\Checkout\Model\Session $checkoutSession,
+        \Magento\Framework\App\Config\ScopeConfigInterface $config,
+        \Magento\Framework\Message\ManagerInterface $messageInterface,
+        \Magento\Framework\Controller\Result\RedirectFactory $redirectFactory,
+        \DevStone\UsageCalculator\Model\ResourceModel\MaxUsage\CollectionFactory $collectionFactory
+    ) {
+        $this->request = $request;
+        $this->customerSession = $session;
+        $this->collectionFactory = $collectionFactory;
+        $this->orderCollectionFactory = $orderCollectionFactory;
+        $this->checkoutSession = $checkoutSession;
+        $this->scopeConfig = $config;
+        $this->messageInterface = $messageInterface;
+        $this->resultRedirectFactory = $redirectFactory;
+    }
+
+    /**
+     * @param \Magento\Checkout\Controller\Cart\Add $subject
+     * @param callable $proceed
+     * @return mixed
+     */
+    public function aroundExecute(\Magento\Checkout\Controller\Cart\Add $subject, callable $proceed)
+    {
+        $usageId = $this->request->getParam('usage_id');
+        if ($this->request->getParam('usage_category') != $this->getCustomLicenseId()) {
+            return $proceed();
+        } elseif (array_key_exists($this->getCustomLicenseId(), $usageId)) {
+            if ($this->customerSession->isLoggedIn()) {
+                /**
+                 * @var \DevStone\UsageCalculator\Model\ResourceModel\MaxUsage\Collection $maxUsageColleciton
+                 */
+                $maxUsageColleciton = $this->collectionFactory->create()
+                    ->addFieldToSelect('max_usage')
+                    ->addFieldToFilter('usage_id', $usageId[$this->getCustomLicenseId()]);
+                if ($maxUsageColleciton->count()) {
+                    $maxUsage = $maxUsageColleciton->getFirstItem()['max_usage'];
+                    if ($maxUsage) {
+                        $totalUsageCountByOrder = $this->getUsageCountByOrders($usageId);
+                        if ($totalUsageCountByOrder < $maxUsage) {
+                            $totalUsageCountByQuote = $this->getUsageCountByQuote($usageId);
+                            if ($totalUsageCountByQuote < $maxUsage) {
+                                return $proceed();
+                            } else {
+                                $this->messageInterface->addErrorMessage(__('We can\'t add this item to your shopping cart right now Becuase you have reached you\'r  max usage limit'));
```

Comment:
Please change this message
```suggestion
                                $this->messageInterface->addErrorMessage(__('You cannot add this item to your cart because this custom license can only be used %1 times', $maxUsage));
```